core/src/model.jl

```diff
@@ -207,9 +211,7 @@ function SciMLBase.step!(model::Model, dt::Float64)::Model
     # If we are at an allocation time, run allocation before the next physical
     # layer timestep. This allows allocation over period (t, t + dt) to use variables
     # set over BMI at time t before calling this function.
-    # Also, don't run allocation at t = 0 since there are no flows yet (#1389).
-    ntimes = t / config.allocation.timestep
-    if t > 0 && round(ntimes) ≈ ntimes
+    if t % config.allocation.timestep ≈ 0
```

Comment:
If `t` is a tiny bit less than a round number of allocation timesteps, this fails:

```
julia> prevfloat(3.0) % 1.0
0.9999999999999996
```

Reply:
Ah yes maybe not risk that ^^ I figured there was a reason you didn't do this before, but I didn't know what it was. I've used this pattern quite a lot though, but I guess mostly for integers